ui/src/common/components/DateRangeAggregation.jsx

```diff
@@ -0,0 +1,119 @@
+import React, { useCallback, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { DatePicker, Row } from 'antd';
+import AggregationBox from './AggregationBox';
+import {
+  DATE_RANGE_FORMAT,
+  RANGE_AGGREGATION_SELECTION_SEPARATOR,
+} from '../constants';
+
+function DateRangeAggregation({ name, onChange, selections }) {
+  const splitSelections = useMemo(
+    () => selections && selections.split(RANGE_AGGREGATION_SELECTION_SEPARATOR),
+    [selections]
+  );
+
+  const startDate = useMemo(
+    () => (splitSelections && splitSelections[0]) || null,
+    [splitSelections]
+  );
+
+  const endDate = useMemo(
+    () => (splitSelections && splitSelections[1]) || null,
+    [splitSelections]
+  );
+
+  const startDateMoment = useMemo(
+    () => startDate && moment.utc(startDate, DATE_RANGE_FORMAT),
+    [startDate]
+  );
+
+  const endDateMoment = useMemo(
+    () => endDate && moment.utc(endDate, DATE_RANGE_FORMAT),
+    [endDate]
+  );
+
+  const disabledStartDate = useCallback(
+    givenStartDate => {
+      if (!givenStartDate || !endDateMoment) {
+        return false;
+      }
+      return givenStartDate.valueOf() > endDateMoment.valueOf();
```

Comment:
you can use `moment` functions like https://momentjs.com/docs/#/query/is-same-or-before/